src/net/sourceforge/kolmafia/persistence/NPCStoreDatabase.java

```diff
@@ -108,6 +109,25 @@ public static final String getStoreName(final String storeId) {
         : NPCStoreDatabase.storeNameById.get(storeId);
   }
 
+  public static final List<NPCPurchaseRequest> getAvailablePurchaseRequests(final int itemId) {
+    List<NPCPurchaseRequest> items = NPCStoreDatabase.NPC_ITEMS.get(itemId);
+    if (items == null || items.size() == 0) {
+      return null;
+    }
+
+    List<NPCPurchaseRequest> result = new ArrayList<>();
+
+    for (var item : items) {
+      boolean canPurchase = canPurchase(item.getStoreId(), item.getShopName(), itemId);
+      item.setCanPurchase(canPurchase);
```

Comment:
if we can't purchase, this modifies the original item. did you want to do that, or only if it gets added to the list?

Reply:
The routine which used to be called - getPurchaseRequest(itemId) - would return the first available purchase request and, if none were available, would return the last found unavailable one - and set it unavailable.

As this is coded, it probably should modify the unavailable ones, although it shouldn't hurt.
But if I wanted (somewhat) similar behavior, should it return a list of the last-found unavailable one?

Reply:
I adjusted it to only set canPurchase for items it will return.
I did not do anything about returning ... something ... if none of the requests are available.
Considering the method is called getAvailablePurchaseRequests, it is behaving as advertised. :)